flink-streaming-java/src/test/java/org/apache/flink/streaming/api/graph/StreamGraphGeneratorBatchExecutionTest.java

```diff
@@ -105,6 +107,7 @@ public void testOneInputTransformation() {
         assertThat(graph.getStateBackend(), instanceOf(BatchExecutionStateBackend.class));
         // the provider is passed as a lambda therefore we cannot assert the class of the provider
         assertThat(graph.getTimerServiceProvider(), notNullValue());
+        assertThat(graph.getJobType(), is(JobType.BATCH));
```

Comment:
I'm not really sold on this shotgun-style assertion approach.

I'd think that this would be fine for a separate test:

```
        StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
        SingleOutputStreamOperator<Integer> process =
                env.fromElements(1, 2).keyBy(Integer::intValue).process(DUMMY_PROCESS_FUNCTION);
        DataStreamSink<Integer> sink = process.addSink(new DiscardingSink<>());
        StreamGraphGenerator graphGenerator =
                new StreamGraphGenerator(
                        Collections.singletonList(sink.getTransformation()),
                        env.getConfig(),
                        env.getCheckpointConfig());
        graphGenerator.setRuntimeExecutionMode(RuntimeExecutionMode.BATCH);

        StreamGraph graph = graphGenerator.generate();
        assertThat(graph.getJobType(), is(JobType.BATCH));
```

Reply:
> I'm not really sold on this shotgun-style assertion approach.

My thinking was the following: All tests seem to assert a bunch of assumptions (batch state backend, timer service set etc.) .. since this is the style of the tests there, I added the BATCH job type everywhere.
I was too lazy adding a separate batch job-type test, since I assumed enough coverage through the other change.

I will follow your suggestion.